homeassistant/components/zwave_js/__init__.py

```diff
@@ -145,44 +196,18 @@ async def async_on_node_ready(node: ZwaveNode) -> None:
         value_updates_disc_info: dict[str, ZwaveDiscoveryInfo] = {}
 
         # run discovery on all node values and create/update entities
-        for disc_info in async_discover_values(node, device):
-            platform = disc_info.platform
-
-            # This migration logic was added in 2021.3 to handle a breaking change to
-            # the value_id format. Some time in the future, this call (as well as the
-            # helper functions) can be removed.
-            async_migrate_discovered_value(
-                hass,
-                ent_reg,
-                registered_unique_ids[device.id][platform],
-                device,
-                client,
-                disc_info,
+        for disc_info in async_discover_node_values(node, device, processed_value_ids):
+            await async_handle_discovery_info(
```

Comment:
Can we `asyncio.gather` this?

Reply:
I get the following when I try, any ideas? Do we have to move away from using a generator?
```python
homeassistant/components/zwave_js/__init__.py:288: error: No overload variant of "gather" matches argument type "Generator[None, None, None]"  [call-overload]
homeassistant/components/zwave_js/__init__.py:288: note: Possible overload variants:
homeassistant/components/zwave_js/__init__.py:288: note:     def gather(coro_or_future1: Union[Future[Any], Generator[Any, None, Any], Awaitable[Any]], coro_or_future2: Union[Future[Any], Generator[Any, None, Any], Awaitable[Any]], coro_or_future3: Union[Future[Any], Generator[Any, None, Any], Awaitable[Any]], coro_or_future4: Union[Future[Any], Generator[Any, None, Any], Awaitable[Any]], coro_or_future5: Union[Future[Any], Generator[Any, None, Any], Awaitable[Any]], coro_or_future6: Union[Future[Any], Generator[Any, None, Any], Awaitable[Any]], *coros_or_futures: Union[Future[Any], Generator[Any, None, Any], Awaitable[Any]], loop: Optional[AbstractEventLoop] = ..., return_exceptions: bool = ...) -> Future[List[Any]]
homeassistant/components/zwave_js/__init__.py:288: note:     def [_T1] gather(coro_or_future1: Union[Future[_T1], Generator[Any, None, _T1], Awaitable[_T1]], *, loop: Optional[AbstractEventLoop] = ..., return_exceptions: Literal[False] = ...) -> Future[Tuple[_T1]]
homeassistant/components/zwave_js/__init__.py:288: note:     <9 more similar overloads not shown, out of 11 total overloads>
```